src/quo/components/wallet/token_input/view.cljs

```diff
@@ -92,36 +84,49 @@
-          [rn/text-input
-           (cond-> {:style                    (style/text-input theme error?)
-                    :placeholder-text-color   (style/placeholder-text theme)
-                    :auto-focus               true
-                    :ref                      set-ref
-                    :placeholder              "0"
-                    :keyboard-type            :numeric
-                    :max-length               12
-                    :on-change-text           handle-on-change-text
-                    :selection-color          customization-color
-                    :show-soft-input-on-focus show-keyboard?
-                    :on-selection-change      handle-selection-change
-                    :selection                (clj->js selection)}
-             controlled-input?       (assoc :value value)
-             (not controlled-input?) (assoc :default-value value-internal))]]
-         [token-label
-          {:theme theme
-           :text  (if crypto? token currency)
-           :value (if controlled-input? value value-internal)}]]))))
+      (let [theme        (quo.theme/use-theme)
+            window       (rn/get-window)
+            window-width (:width window)]
+        [rn/pressable
+         {:style    {:width          "100%"
+                     :flex-direction :row}
+          :on-press on-token-press}
+         [token/view
+          {:token token
+           :size  :size-32}]
+         [rn/view {:style (style/input-container window-width)}
+          [rn/pressable
+           {:style    style/text-input-container
+            :on-press focus-input}
+           [rn/text-input
+            (cond-> {:style                    (style/text-input theme error?)
+                     :placeholder-text-color   (style/placeholder-text theme)
+                     :auto-focus               true
+                     :ref                      set-ref
+                     :placeholder              "0"
+                     :keyboard-type            :numeric
+                     :on-change-text           handle-on-change-text
+                     :selection-color          customization-color
+                     :show-soft-input-on-focus show-keyboard?
+                     :on-selection-change      handle-selection-change
+                     :selection                (clj->js selection)}
+              controlled-input?       (assoc :value value)
+              (not controlled-input?) (assoc :default-value value-internal))]
+           [token-label
+            {:theme theme
+             :text  (if crypto? token currency)
+             :value (if controlled-input? value value-internal)}]]]
+         [button/button
+          {:icon                true
+           :icon-only?          true
+           :size                32
+           :on-press            handle-on-swap
+           :type                :outline
+           :accessibility-label :reorder}
+          :i/reorder]]))))
```

Comment:
Moved button to this view, not sure why it was outside of it, but refactored to fix `token-label` overflowing when the input is too large.